ksp_plugin_test/orbit_analyser_test.cpp

```diff
@@ -0,0 +1,137 @@
+﻿
+#include "ksp_plugin/orbit_analyser.hpp"
+
+#include "astronomy/standard_product_3.hpp"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "ksp_plugin/integrators.hpp"
+#include "physics/solar_system.hpp"
+#include "testing_utilities/approximate_quantity.hpp"
+#include "testing_utilities/is_near.hpp"
+
+namespace principia {
+namespace ksp_plugin {
+
+using astronomy::ITRS;
+using astronomy::OrbitRecurrence;
+using astronomy::StandardProduct3;
+using base::not_null;
+using geometry::Position;
+using integrators::SymmetricLinearMultistepIntegrator;
+using integrators::methods::QuinlanTremaine1990Order12;
+using physics::BodySurfaceDynamicFrame;
+using physics::Ephemeris;
+using physics::RotatingBody;
+using physics::SolarSystem;
+using quantities::astronomy::JulianYear;
+using quantities::astronomy::TerrestrialEquatorialRadius;
+using quantities::si::Hour;
+using quantities::si::Kilo;
+using quantities::si::Metre;
+using quantities::si::Milli;
+using quantities::si::Radian;
+using testing_utilities::IsNear;
+using testing_utilities::operator""_⑴;
+using ::testing::AllOf;
+using ::testing::Eq;
+using ::testing::IsNull;
+using ::testing::Optional;
+using ::testing::Property;
+
+class OrbitAnalyserTest : public testing::Test {
+ protected:
+  OrbitAnalyserTest()
+      : earth_1957_(RemoveAllButEarth(SolarSystem<Barycentric>(
+            SOLUTION_DIR / "astronomy" / "sol_gravity_model.proto.txt",
+            SOLUTION_DIR / "astronomy" /
+                "sol_initial_state_jd_2436116_311504629.proto.txt",
+            /*ignore_frame=*/true))),
+        // As there is only one body left in this ephemeris, integration is
+        // exact up to roundoff error regardless of the step size.  Use a long
+        // step so that we do not waste time computing the ephemeris.
+        ephemeris_(earth_1957_.MakeEphemeris(
+            /*accuracy_parameters=*/{/*fitting_tolerance=*/1 * Milli(Metre),
+                                     /*geopotential_tolerance=*/0x1p-24},
+            Ephemeris<Barycentric>::FixedStepParameters(
+                SymmetricLinearMultistepIntegrator<QuinlanTremaine1990Order12,
+                                                   Position<Barycentric>>(),
+                /*step=*/1 * JulianYear))),
+        earth_(*earth_1957_.rotating_body(*ephemeris_, "Earth")),
+        itrs_(ephemeris_.get(), &earth_),
+        topex_poséidon_(SOLUTION_DIR / "astronomy" / "standard_product_3" /
+                            "grgtop03.b97344.e97348.D_S.sp3",
+                        StandardProduct3::Dialect::GRGS) {}
+
+  SolarSystem<Barycentric> earth_1957_;
+  not_null<std::unique_ptr<Ephemeris<Barycentric>>> ephemeris_;
+  RotatingBody<Barycentric> const& earth_;
+  BodySurfaceDynamicFrame<Barycentric, ITRS> itrs_;
+  StandardProduct3 topex_poséidon_;
+
+ private:
+  SolarSystem<Barycentric> RemoveAllButEarth(
+      SolarSystem<Barycentric> solar_system) {
```

Comment:
Could we `const&` this?  Or else pass by `&`.

Reply:
No, to both.

`SolarSystem` is move-only; if you look at the callsite, the parameter is a temporary; a reference would not work.